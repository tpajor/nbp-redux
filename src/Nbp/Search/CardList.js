import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';

const CardList = ({ cards, signedIn, userSignedIn, deleteCard, showDetails }) => {
  return (
    <div>
      {cards.map(card => 
        <Card key={card.id}
          signedIn={signedIn}
          userSignedIn={userSignedIn}
          deleteCard={deleteCard}
          cardData={card}
          isTemporary={false}
          showDetails={showDetails}
          cards={cards}
        />
      ).reverse()}
    </div>
  );
};

CardList.propTypes = {
  cards: PropTypes.array,
  deleteCard: PropTypes.func,
  showDetails: PropTypes.func,
  signedIn: PropTypes.bool,
  userSignedIn: PropTypes.string,
};

export default CardList;