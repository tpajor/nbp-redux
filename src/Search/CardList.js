import React from 'react';
import uuidv4 from 'uuid/v4';
import Card from './Card';

const CardList = (props) => {
  return (
    <div>
      {props.cards.map(card => 
        <Card key={uuidv4()}
          deleteCard={props.deleteCard}
          cardData={card}
          isTemporary={false}
          showDetails={props.showDetails}
        />
      ).reverse()}
    </div>
  );
};

export default CardList;