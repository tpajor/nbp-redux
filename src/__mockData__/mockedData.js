const card = { id: 'someId', code: 'USD', currency: 'dolar amerykański', rates: [{ ask: 3.3, bid: 3.4 }]};
const card2 = { id: 'someId2', code: 'USD', currency: 'dolar amerykański', rates: { ask: 3.3, bid: 3.4 }};

const search = {
  cards: [card, card2],
  temporaryCard: card,
  searchError: false,
  addCardError: true,
  addCardErrorMessage: 'error',
  currenciesTable: ['usd', 'chf', 'aud'],
  currencyCodeToViewInDetail: 'usd',
};

const user = { userName: 'User', pass: 'pass' };
const user2 = { userName: 'User2', pass: 'pass2'};

export const login = {
  signingError: true,
  signingErrorMessage: 'error',
  signedIn: true,
  userSignedIn: 'User',
  users: [user, user2]
};

const lastRates = [
  { ask: 3.5983, bid: 3.5271, effectiveDate: "2018-05-07", no: "087/C/NBP/2018" },
  { ask: 3.6088, bid: 3.5374, effectiveDate: "2018-05-08", no: "088/C/NBP/2018" }
];

const detail = {
  lastRates: lastRates,
  detailsInputError: false,
  detailsInputErrorMessage: '',
};

export const initialState = { search, login, detail };

export const searchViewData = {
  cards: search.cards,
  temporaryCard: search.temporaryCard,
  searchError: search.searchError,
  signedIn: login.signedIn,
  userSignedIn: login.userSignedIn,
  addCardError: search.addCardError,
  addCardErrorMessage: search.addCardErrorMessage,
  currenciesTable: search.currenciesTable,
};

export const detailViewData = {
  currencyCodeToViewInDetail: search.currencyCodeToViewInDetail,
  lastRates: detail.lastRates,
  detailsInputError: detail.detailsInputError,
  detailsInputErrorMessage: detail.detailsInputErrorMessage,
  signedIn: login.signedIn,
};