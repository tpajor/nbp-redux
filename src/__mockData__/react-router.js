const Link = () => {
  return jest.fn();
}

export default Link;