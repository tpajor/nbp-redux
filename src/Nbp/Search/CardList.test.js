import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CardList from './CardList';
import { searchViewData } from '../../__mockData__/mockedData';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    cards: searchViewData.cards,
    signedIn: searchViewData.signedIn,
    userSignedIn: searchViewData.userSignedIn,
    deleteCard: jest.fn(),
    showDetails: jest.fn(),
  };

  const enzymeWrapper = shallow(<CardList {...props} />);

  return enzymeWrapper;
};

describe('CardList component', () => {
  it('should render self and subcomponents with their props', () => {
    const enzymeWrapper = setup();

    expect(enzymeWrapper.length).toBe(1);

    expect(enzymeWrapper.find('div').length).toBe(1);

    expect(enzymeWrapper.find('Card').length).toEqual(searchViewData.cards.length);
    expect(enzymeWrapper.find('Card').first().props().signedIn).toEqual(searchViewData.signedIn);
    expect(enzymeWrapper.find('Card').first().props().userSignedIn).toEqual(searchViewData.userSignedIn);
    expect(enzymeWrapper.find('Card').first().props().isTemporary).toBe(false);
    expect(enzymeWrapper.find('Card').first().props().cardData).toEqual(searchViewData.cards[1]);
    expect(enzymeWrapper.find('Card').first().props().cards).toEqual(searchViewData.cards);
    expect(typeof enzymeWrapper.find('Card').first().props().deleteCard).toBe('function');
    expect(typeof enzymeWrapper.find('Card').first().props().showDetails).toBe('function');
  });

  it('should render correctly', () => {
    const enzymeWrapper = setup();
    expect(enzymeWrapper).toMatchSnapshot();
  });
});