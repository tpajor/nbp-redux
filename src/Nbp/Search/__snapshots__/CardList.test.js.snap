// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    cards={
        Array [
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId2",
              "rates": Object {
                "ask": 3.3,
                "bid": 3.4,
              },
            },
          ]
    }
    deleteCard={[Function]}
    showDetails={[Function]}
    signedIn={true}
    userSignedIn="User"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardData={
                    Object {
                              "code": "USD",
                              "currency": "dolar amerykański",
                              "id": "someId2",
                              "rates": Object {
                                "ask": 3.3,
                                "bid": 3.4,
                              },
                            }
          }
          cards={
                    Array [
                              Object {
                                "code": "USD",
                                "currency": "dolar amerykański",
                                "id": "someId",
                                "rates": Array [
                                  Object {
                                    "ask": 3.3,
                                    "bid": 3.4,
                                  },
                                ],
                              },
                              Object {
                                "code": "USD",
                                "currency": "dolar amerykański",
                                "id": "someId2",
                                "rates": Object {
                                  "ask": 3.3,
                                  "bid": 3.4,
                                },
                              },
                            ]
          }
          deleteCard={[Function]}
          isTemporary={false}
          showDetails={[Function]}
          signedIn={true}
          userSignedIn="User"
/>,
        <Card
          cardData={
                    Object {
                              "code": "USD",
                              "currency": "dolar amerykański",
                              "id": "someId",
                              "rates": Array [
                                Object {
                                  "ask": 3.3,
                                  "bid": 3.4,
                                },
                              ],
                            }
          }
          cards={
                    Array [
                              Object {
                                "code": "USD",
                                "currency": "dolar amerykański",
                                "id": "someId",
                                "rates": Array [
                                  Object {
                                    "ask": 3.3,
                                    "bid": 3.4,
                                  },
                                ],
                              },
                              Object {
                                "code": "USD",
                                "currency": "dolar amerykański",
                                "id": "someId2",
                                "rates": Object {
                                  "ask": 3.3,
                                  "bid": 3.4,
                                },
                              },
                            ]
          }
          deleteCard={[Function]}
          isTemporary={false}
          showDetails={[Function]}
          signedIn={true}
          userSignedIn="User"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "someId2",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "code": "USD",
            "currency": "dolar amerykański",
            "id": "someId2",
            "rates": Object {
              "ask": 3.3,
              "bid": 3.4,
            },
          },
          "cards": Array [
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId2",
              "rates": Object {
                "ask": 3.3,
                "bid": 3.4,
              },
            },
          ],
          "deleteCard": [Function],
          "isTemporary": false,
          "showDetails": [Function],
          "signedIn": true,
          "userSignedIn": "User",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "someId",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "code": "USD",
            "currency": "dolar amerykański",
            "id": "someId",
            "rates": Array [
              Object {
                "ask": 3.3,
                "bid": 3.4,
              },
            ],
          },
          "cards": Array [
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId2",
              "rates": Object {
                "ask": 3.3,
                "bid": 3.4,
              },
            },
          ],
          "deleteCard": [Function],
          "isTemporary": false,
          "showDetails": [Function],
          "signedIn": true,
          "userSignedIn": "User",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardData={
                        Object {
                                    "code": "USD",
                                    "currency": "dolar amerykański",
                                    "id": "someId2",
                                    "rates": Object {
                                      "ask": 3.3,
                                      "bid": 3.4,
                                    },
                                  }
            }
            cards={
                        Array [
                                    Object {
                                      "code": "USD",
                                      "currency": "dolar amerykański",
                                      "id": "someId",
                                      "rates": Array [
                                        Object {
                                          "ask": 3.3,
                                          "bid": 3.4,
                                        },
                                      ],
                                    },
                                    Object {
                                      "code": "USD",
                                      "currency": "dolar amerykański",
                                      "id": "someId2",
                                      "rates": Object {
                                        "ask": 3.3,
                                        "bid": 3.4,
                                      },
                                    },
                                  ]
            }
            deleteCard={[Function]}
            isTemporary={false}
            showDetails={[Function]}
            signedIn={true}
            userSignedIn="User"
/>,
          <Card
            cardData={
                        Object {
                                    "code": "USD",
                                    "currency": "dolar amerykański",
                                    "id": "someId",
                                    "rates": Array [
                                      Object {
                                        "ask": 3.3,
                                        "bid": 3.4,
                                      },
                                    ],
                                  }
            }
            cards={
                        Array [
                                    Object {
                                      "code": "USD",
                                      "currency": "dolar amerykański",
                                      "id": "someId",
                                      "rates": Array [
                                        Object {
                                          "ask": 3.3,
                                          "bid": 3.4,
                                        },
                                      ],
                                    },
                                    Object {
                                      "code": "USD",
                                      "currency": "dolar amerykański",
                                      "id": "someId2",
                                      "rates": Object {
                                        "ask": 3.3,
                                        "bid": 3.4,
                                      },
                                    },
                                  ]
            }
            deleteCard={[Function]}
            isTemporary={false}
            showDetails={[Function]}
            signedIn={true}
            userSignedIn="User"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "someId2",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId2",
              "rates": Object {
                "ask": 3.3,
                "bid": 3.4,
              },
            },
            "cards": Array [
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId",
                "rates": Array [
                  Object {
                    "ask": 3.3,
                    "bid": 3.4,
                  },
                ],
              },
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId2",
                "rates": Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              },
            ],
            "deleteCard": [Function],
            "isTemporary": false,
            "showDetails": [Function],
            "signedIn": true,
            "userSignedIn": "User",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "someId",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
            "cards": Array [
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId",
                "rates": Array [
                  Object {
                    "ask": 3.3,
                    "bid": 3.4,
                  },
                ],
              },
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId2",
                "rates": Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              },
            ],
            "deleteCard": [Function],
            "isTemporary": false,
            "showDetails": [Function],
            "signedIn": true,
            "userSignedIn": "User",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
